internal/store/queuestore.go

```diff
@@ -107,9 +107,9 @@ func (store *QueueStore[_]) Delete() error {
 }
 
 // PutMultiple - puts an item to the store.
-func (store *QueueStore[I]) PutMultiple(item []I) error {
+func (store *QueueStore[I]) PutMultiple(items []I) error {
```

Comment:
It would be nice to see writes return the generated Key(uid). 

This would be allow me to use this storage module for the audit logging as well.